src/overlays/actors/ovl_Obj_Lift/z_obj_lift.c

```diff
@@ -28,29 +36,188 @@ const ActorInit Obj_Lift_InitVars = {
+    ICHAIN_F32_DIV1000(gravity, -600, ICHAIN_CONTINUE),   ICHAIN_F32_DIV1000(minVelocityY, -15000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 2000, ICHAIN_CONTINUE), ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 2000, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lift/ObjLift_Draw.s")
+static f32 sScales[] = { 0.1f, 0.05f };
+static f32 sMaxFallDistances[] = { -18.0f, -9.0f };
+
+void ObjLift_SetupAction(ObjLift* this, ObjLiftActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void ObjLift_InitDynaPoly(ObjLift* this, GlobalContext* globalCtx, CollisionHeader* collision, DynaPolyMoveFlag flags) {
+    s32 pad;
+    CollisionHeader* colHeader = NULL;
+    s32 pad2;
+
+    DynaPolyActor_Init(&this->dyna, flags);
+    CollisionHeader_GetVirtual(collision, &colHeader);
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+    if (this->dyna.bgId == BG_ACTOR_MAX) {
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_obj_lift.c", 188,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void func_80B96160(ObjLift* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f* temp_s3;
+    s32 pad0;
+    s32 i;
+
+    temp_s3 = &this->dyna.actor.world.pos;
+
+    for (i = 0; i < ARRAY_COUNT(sFragmentScales); i++) {
+        pos.x = sFragmentScales[i].x * this->dyna.actor.scale.x + temp_s3->x;
+        pos.y = temp_s3->y;
+        pos.z = sFragmentScales[i].z * this->dyna.actor.scale.z + temp_s3->z;
+        velocity.x = sFragmentScales[i].x * this->dyna.actor.scale.x * 0.8f;
+        velocity.y = Rand_ZeroOne() * 10.0f + 6.0f;
+        velocity.z = sFragmentScales[i].z * this->dyna.actor.scale.z * 0.8f;
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, temp_s3, -256, (Rand_ZeroOne() < 0.5f) ? 64 : 32, 15, 15, 0,
+                             (Rand_ZeroOne() * 50.0f + 50.0f) * this->dyna.actor.scale.x, 0, 32, 50, KAKERA_COLOR_NONE,
+                             OBJECT_D_LIFT, gCollapsingPlatformDL);
+    }
+
+    if (!((this->dyna.actor.params >> 1) & 1)) {
+        func_80033480(globalCtx, &this->dyna.actor.world.pos, 120.0f, 12, 120, 100, 1);
+    } else if (((this->dyna.actor.params >> 1) & 1) == 1) {
+        func_80033480(globalCtx, &this->dyna.actor.world.pos, 60.0f, 8, 60, 100, 1);
+    }
+}
+
+void ObjLift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjLift* this = THIS;
+
+    ObjLift_InitDynaPoly(this, globalCtx, &gCollapsingPlatformCol, DPM_PLAYER);
+
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params >> 2) & 0x3F)) {
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+
+    Actor_SetScale(&this->dyna.actor, sScales[(this->dyna.actor.params >> 1) & 1]);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->unk168.x = Rand_ZeroOne() * 65535.5f;
```

Comment:
`unk168` is used as a rotation so hex for binangles is prefered (or maybe not idk anymore)

https://github.com/StardustSauce/oot/pull/2

Reply:
these cant be hex, its 65535.5 lol, pretty weird

Reply:
going to leave for now.
I understand angles are a bit confusing at the moment. For now we are leaving it to the judgement of whoever did the file. Its clear the devs were not consistent themselves about hex vs decimal. So there are contexts where both can make sense over the other

we will eventually have a larger discussion about this because some people do think it should just be hex always. It will need to be cleaned up later if we decide on something else. But for now consistency is not required